activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -500,7 +500,9 @@ def replace(other_array)
       #   Pet.find(1, 2, 3)
       #   # => ActiveRecord::RecordNotFound: Couldn't find all Pets with 'id': (1, 2, 3)
       def delete_all(dependent = nil)
-        @association.delete_all(dependent)
+        count = @association.delete_all(dependent)
```

Comment:
`@association.delete_all(dependent)` doesn't return count (not guaranteed, almost nil).
How about using `@association.delete_all(dependent).tap { reset_scope }` to not care about what is returning?